actions/install/installer.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# SLSA verifier install script
+
+set -euo pipefail
+shopt -s expand_aliases
+alias log_info="echo \"INFO:\""
+alias log_error="echo \"ERROR:\""
+
+verifier_release="$1"
+install_dir="$HOME/.slsa/bin/${verifier_release}"
+
+mkdir -p "$install_dir"
+cd "$install_dir"
+
+if [[ "$verifier_release" == "main" ]]; then
+    log_info "installing verifier via 'go install' from its main version"
+    gobin="$(go env GOPATH)/bin"
+    go install github.com/slsa-framework/slsa-verifier/cli/slsa-verifier@main
+    ln -s "$gobin/slsa-verifier" "$install_dir/slsa-verifier"
+    echo "Installed SLSA verifier at $install_dir/slsa-verifier"
+    exit 0
+fi
+
+# Download v1.2.0 for bootstrapping
+binary_addr="https://github.com/slsa-framework/slsa-verifier/releases/download/v1.2.0/slsa-verifier-linux-amd64"
+sha256="37db23392c7918bb4e243cdb097ed5f9d14b9b965dc1905b25bc2d1c0c91bf3d slsa-verifier-linux-amd64"
+if [[ "$verifier_release" == "test-force-checksum-error" ]]; then
+  binary_addr="https://github.com/slsa-framework/slsa-verifier/releases/download/v0.0.1/slsa-verifier-linux-amd64"
+fi
+curl -sL "$binary_addr" -o slsa-verifier-linux-amd64
+echo "$sha256" >> SHA256SUM.md
+
+if ! sha256sum --quiet -c SHA256SUM.md ; then
+  rm slsa-verifier-linux-amd64
```

Comment:
Hw about creating a temporary folder with all files needed, and do a `rm *` at the end? This way we don't have to worry about which files we need to remove based on how far the script has executed. We could also have a `cleanup` function to do it. Wdut?

Reply:
Yes! or all work in a tmp dir, with only a final success moved